services/sn_analysis_service/app.py

```diff
@@ -22,38 +29,48 @@
 matplotlib.use('Agg')
 rng = np.random.default_rng()
 
-app = Quart(__name__)
-app.debug = False
-
 default_analysis_parameters = {"source": "nugent-sn2p", "fix_z": False}
 
 
-def upload_analysis_results(results, data_dict):
+def upload_analysis_results(results, data_dict, request_timeout=60):
     """
     Upload the results to the webhook.
-
-    TO BE COMPLETED
     """
-    log("upload results to webhook")
-    assert isinstance(results, dict)
-    assert isinstance(data_dict, dict)
-    pass
 
+    log("Uploading results to webhook")
+    if data_dict["callback_method"] != "POST":
+        log("Callback URL is not a POST URL. Skipping.")
+        return
+    url = data_dict["callback_url"]
+    try:
+        _ = requests.post(
+            url,
+            json=results,
+            timeout=request_timeout,
+        )
+    except requests.exceptions.Timeout:
+        # If we timeout here then it's precisely because
+        # we cannot write back to the SkyPortal instance.
+        # So returning something doesn't make sense in this case.
+        # Just log it and move on...
+        log("Callback URL timedout. Skipping.")
+    except Exception as e:
+        log(f"Callback exception {e}.")
 
-async def run_sn_model(dd):
+
+def run_sn_model(dd):
     """
     Use `sncosmo` to fit data to a model with name `source_name`.
 
-    We expect the `inputs` dictionary to have the following keys:
+    For this analysis, we expect the `inputs` dictionary to have the following keys:
        - source: the name of the model to fit to the data
        - fix_z: whether to fix the redshift
        - photometry: the photometry to fit to the model (in csv format)
        - redshift: the known redshift of the object
 
-    TODO in the next PR: send the results back to the webhook
-
+    Other analysis services may require additional keys in the `inputs` dictionary.
     """
-    data_dict = await dd
+    data_dict = dd
```

Comment:
`dd` is rather obscure; why not just use `data_dict` as the argument name?

Reply:
fixed.